### PR TITLE
fix: use innerText instead of innerHTML for translations

### PR DESCRIPTION
**Pull request by dunglas**
Fixes: https://github.com/dunglas/vaccin.click/pull/91#issuecomment-995930345

WDYT @hschaeidt?

### CODE REVIEW

**Comment by julienw** [comment]
Doing innerHTML directly from translations sounds a bad idea indeed.
But splitting looks straightforward for the second (because the <a> wraps the full message) but more difficult for the first IMO.

I don't have a good suggestion TBH :/

**Comment by hschaeidt** [comment]
I agree that this was probably not the best of all possible solutions, but the cheapest. In this specific case, however, we have full control of what comes in the translations and what goes out, so I estimate the security risk as not higher than if anybody pushes malicious code directly into the repo. Does my reasoning make sense here, or did I miss something?

I will work on a clean version where we split out the HTML from the translations, for the sake of sanity. Not sure if we should consider this as a release blocker. What is your opinion?

**Comment by hschaeidt** [comment]
Oh. I didn't understand that this is blocking the release at Firefox's extensions store. I will come up with a workaround tonight! Ignore my previous message and thanks for bringing it up!

**Comment by julienw** [comment]
I believe this is only a warning, where you have to send the source code to the AMO website.

**Comment by julienw** [comment]
I've been told that you could use dompurify to sanitize the HTML, and that's accepted by the AMO reviewers.

**Comment by hschaeidt** [comment]
@julienw that sounds like a good quick fix/workaround we potentially need here.

My initial thought was to consider how we would solve this issue using modern frontend frameworks. In this case, this text component could have been extracted into a separate component that takes a URL string as a property. Maybe we can do the same here by using a web component for this very specific case? What do you think?

**Comment by hschaeidt** [comment]
The quickfix is ready. It really isn't famous and probably not a sustainable solution, but now we have time to think about how to tackle this topic in a sustainable way.

- `npx web-ext lint` warnings are fixed with dunglas's commit
- language dependent urls can be defined in the html via custom data attributes (`data-i18n-href`)

**Comment by dunglas** [comment]
Shouldn't we put the internationalized URLs directly in the message catalog?

**Comment by hschaeidt** [comment]
I am not sure if I understood your point, is this is what you were looking for:

- https://github.com/dunglas/vaccin.click/pull/94/files#diff-4fee12151ea49a5007142ae55129100fd236810c59ca5430e407ed62e49f289aR59
- https://github.com/dunglas/vaccin.click/pull/94/files#diff-4fee12151ea49a5007142ae55129100fd236810c59ca5430e407ed62e49f289aR71

The URL remaining in the `index.html` acts just as a default and to have valid HTML there.

Or did you mean to put them there, following another naming schema, to make them easier identifiable as URLs?

File: browser_action/index.html [hunk 1]
```diff
@@ -105,10 +105,15 @@
...
-            >Réserver automatiquement (vous devez être
-            <a href="https://www.doctolib.fr/sessions/new" target="_blank"
-              >connecté à Doctolib</a
+          <label for="enableAutoBook"
+            ><span data-i18n="enableAutoBook">Réserver automatiquement</span>
+            <a
+              data-i18n-href="doctolibLoginLink"
+              href="https://www.doctolib.fr/sessions/new"
+              target="_blank"
+              >(<span data-i18n="enableAutoBookHint"
```
Comment (julienw) [inline browser_action/index.html:114]: I'd put the parenthesis inside the localized string

**Comment by dunglas** [comment]
LGTM

**Pull request merged**